api/server/router/build/build.go

```diff
@@ -5,14 +5,13 @@ import "github.com/docker/docker/api/server/router"
 // buildRouter is a router to talk with the build controller
 type buildRouter struct {
 	backend Backend
+	daemon  daemonBackend
 	routes  []router.Route
 }
 
 // NewRouter initializes a new build router
-func NewRouter(b Backend) router.Router {
-	r := &buildRouter{
-		backend: b,
-	}
+func NewRouter(b Backend, d daemonBackend) router.Router {
```

Comment:
I would rather avoid boolean arguments, they are a code smell.

I could move `HasExperimental()` to Backend and expose it from the new api/server/backend instead.